relay/relay.go

```diff
@@ -1060,6 +1060,10 @@ func (r *Relay) setSyncConfig() error {
 // AdjustGTID implements Relay.AdjustGTID
 // starting sync at returned gset will wholly fetch a binlog from beginning of the file.
 func (r *Relay) adjustGTID(ctx context.Context, gset gtid.Set) (gtid.Set, error) {
+	failpoint.Inject("MockAdjustGTIDReturnEmptyGTID", func(_ failpoint.Value) {
+		gset, _ = gtid.ParserGTID("mysql", "")
```

Comment:
seems this function will not shrink gset

Reply:
This actually shrinks the gtidset (when the previous gitd set is empty), but i should mock `GetPreviousGTIDFromGTIDSet` instead of `adjustGTID` added in https://github.com/pingcap/dm/pull/2100/commits/84a4ebec327f432443098cf993a71e6132e6d9ff